File: go/api/DDns.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Ddns 对应 DDns.py。
type Ddns struct {
	*session.ApiGroup
}

// NewDdns 创建 DDns API 分组。
func NewDdns(s *session.Session) *Ddns {
	return &Ddns{ApiGroup: session.NewApiGroup(s)}
}

// GetDdnsList 对应 ddns/ddns-list。
func (d *Ddns) GetDdnsList() (map[string]interface{}, error) {
	return d.S.Get("ddns/ddns-list", nil, "api")
}

// GetStatus 对应 ddns/status。
func (d *Ddns) GetStatus() (map[string]interface{}, error) {
	return d.S.Get("ddns/status", nil, "api")
}

// Serverlist 对应 ddns/serverlist。
func (d *Ddns) Serverlist() (map[string]interface{}, error) {
	return d.S.Get("ddns/serverlist", nil, "api")
}

File: src/api/d_dns.rs
//! DDns API group (`api/DDns.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// DDns API group.
pub struct DDns<'a> {
    conn: &'a Connection,
}

impl<'a> DDns<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        DDns { conn }
    }

    /// `ddns/ddns-list`. List of DDNS configurations.
    pub fn get_ddns_list(&self) -> Result<Value> {
        get_value(self.conn, "api/ddns/ddns-list")
    }

    /// `ddns/status`. DDNS status.
    pub fn get_status(&self) -> Result<Value> {
        get_value(self.conn, "api/ddns/status")
    }

    /// `ddns/serverlist`. List of DDNS servers.
    pub fn serverlist(&self) -> Result<Value> {
        get_value(self.conn, "api/ddns/serverlist")
    }
}
